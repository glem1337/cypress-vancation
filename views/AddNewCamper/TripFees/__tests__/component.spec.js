import React from 'react';
import { shallow } from 'enzyme';

import TripFeesComponent from '../component';

describe('TripFeesComponent component tests', () => {
  const props = {
    values: {
      cleaningAndPreparationFee: '',
      mileage: {
        mode: 'unlimited',
        included: 150,
        overage: 0.5,
      },
      generator: {
        hasGenerator: false,
        mode: 'unlimited',
        included: 4,
        overage: 5,
      },
      customFees: {},
    },
    onMilesModeChange: jest.fn(),
    handleSubmit: jest.fn(),
    onBackButtonClick: jest.fn(),
    onGeneratorExistenceChange: jest.fn(),
    onGeneratorModeChange: jest.fn(),
    addCustomFee: jest.fn(),
    removeCustomFee: jest.fn(),
    onFrequencyChanged: jest.fn(),
    isLoading: false,
    isCamperExist: true,
    camperCompleteness: 342,
  };

  const component = shallow(<TripFeesComponent {...props} />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when camper does not exist', () => {
    component.setProps({ isCamperExist: false });

    expect(component).toMatchSnapshot();
  });
});
