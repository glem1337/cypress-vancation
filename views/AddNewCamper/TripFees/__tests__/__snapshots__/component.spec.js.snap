// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripFeesComponent component tests matches snapshot 1`] = `
<Fragment>
  <div
    className="main-listing-container"
  >
    <div
      className="mb-16 mb-md-24"
    >
      <ProgressBar
        content={
          <Memo(FormattedMessage)
            id="addNewCamper.ProgressBar"
          />
        }
        percent={342}
      />
    </div>
    <ForwardRef(InternalForm)
      layout="vertical"
    >
      <CleaningsSection
        cleaningAndPreparationFee=""
      />
      <Divider
        className="mt-24"
      />
      <Row>
        <Col
          span={24}
        >
          <MileageSection
            mode="unlimited"
            onMilesModeChange={[MockFunction]}
          />
        </Col>
      </Row>
      <Divider
        className="mt-24"
      />
      <Row>
        <Col
          span={24}
        >
          <GeneratorSection
            hasGenerator={false}
            mode="unlimited"
            onGeneratorExistenceChange={[MockFunction]}
            onGeneratorModeChange={[MockFunction]}
          />
        </Col>
      </Row>
      <Divider
        className="mt-24"
      />
      <CustomFeesSection
        addCustomFee={[MockFunction]}
        customFees={Array []}
        onFrequencyChanged={[MockFunction]}
        removeCustomFee={[MockFunction]}
      />
    </ForwardRef(InternalForm)>
    <withRouter(Connect(EstimatedEarningCard))
      isSlim={true}
    />
  </div>
  <AddNewCamperBtnForm
    canSave={true}
    isSubmitting={false}
    onBackClick={[MockFunction]}
    onSaveClick={[MockFunction]}
    showGradientButton={true}
    withBackBtn={true}
  />
</Fragment>
`;

exports[`TripFeesComponent component tests matches snapshot when camper does not exist 1`] = `
<div
  className="main-listing-container"
>
  <div
    className="mb-16 mb-md-24"
  >
    <ProgressBar
      content={
        <Memo(FormattedMessage)
          id="addNewCamper.ProgressBar"
        />
      }
      percent={342}
    />
  </div>
  <Row>
    <Col
      span={24}
    >
      <Skeleton
        active={true}
        avatar={false}
        paragraph={true}
        title={true}
      />
    </Col>
  </Row>
</div>
`;
